src/graphql/__tests__/mutations.js

```diff
@@ -0,0 +1,88 @@
+'use strict' 
+
+const EasyGraphQLTester = require('easygraphql-tester')
+const fs = require('fs')
+const path = require('path')
+
+const schema = fs.readFileSync(path.join(__dirname, '..', 'schema.graphql'), 'utf8')
+
+describe('Test GraphQL mutations', () => {
+  let tester
+  
+  beforeAll(() => {
+    tester = new EasyGraphQLTester(schema)
+  })
+
+  describe('Mutations', () => {
+    test('Should pass a valid createContributor', () => {
+      const mutation = `
+        mutation CREATE_CONTRIBUTOR($input: CreateContributorInput) {
+          createContributor(input: $input) {
+            githubUsername
+            email
+            github {
+              username
+              contributionCount
+              pullRequests {
+                id
+                title
+                url
+                number
+                labels {
+                  name
+                  url
+                }
+              }
+            }
+            shopify {
+              id
+              codes {
+                code
+                used
+              }
+            }
+          }
+        }
+      `
```

Comment:
Does this mean that every time the schema changes, we have to update the queries here as well?

Reply:
Yes! this will prevent to break anything else, on the front end and on the backend as well... what I actually do, is to test the same queries/mutations that I had on my front end, that way when I change my schema I'm sure that my FE  will continue working without breaking anything!

Reply:
To play devil's advocate here, does this add any benefit in addition to the tests we already have? 

For example: https://github.com/gatsbyjs/api.gatsbyjs.org/blob/679ea4ef0f6f24d650d4ad765ca58c8be7fd403a/src/graphql/__tests__/resolvers.js#L248-L271

This test (along with the others) validates that the resolvers do what we expect them to do. Are we duplicating effort by adding these tests?

Reply:
Yes, it does!! Actually with this test you’re not testing the resolvers, you’re testing the queries and mutations... so, why is this important? Try changing one field on the schema, the tests that you actually have, are testing the resolvers as you said, so for example: if you rename on the schema a field that was called fullName to name... the test of the resolvers is gonna pass without problem, the logic inside it is valid, but actually you’re returning a property that is invalid(and you’re not testing it), so if you try to get it on your query is not going to work as expected, there should be an error because you are waiting name but right now it is sending fullName as it was!!